pkgs/applications/networking/instant-messengers/wavebox/default.nix

```diff
@@ -1,94 +1,153 @@
+
+in stdenv.mkDerivation (finalAttrs: {
   pname = "wavebox";
-  version = "10.124.17-2";
+  version = "10.126.14-2";
 
   src = fetchurl {
-    url = "https://download.wavebox.app/stable/linux/tar/Wavebox_${version}.tar.gz";
-    sha256 = "sha256-RS1/zs/rFWsj29BrT8Mb2IXgy9brBsQypxfvnd7pKl0=";
+    url = "https://download.wavebox.app/stable/linux/deb/amd64/wavebox_${finalAttrs.version}_amd64.deb";
+    hash = "sha256-btd+QGgRNOaR7g4/y6ccmGtR1wUaheoDT96pIkxYdA4=";
   };
 
-  # don't remove runtime deps
-  dontPatchELF = true;
-  # ignore optional Qt 6 shim
-  autoPatchelfIgnoreMissingDeps = [ "libQt6Widgets.so.6" "libQt6Gui.so.6" "libQt6Core.so.6" ];
-
-  nativeBuildInputs = [ autoPatchelfHook makeWrapper qt5.wrapQtAppsHook copyDesktopItems ];
-
-  buildInputs = with xorg; [
-    libXdmcp
-    libXScrnSaver
-    libXtst
-    libxshmfence
-    libXdamage
-  ] ++ [
-    alsa-lib
-    gtk3
-    nss
-    libdrm
-    mesa
-    gtk4
-    qt5.qtbase
+  nativeBuildInputs = [ patchelf makeWrapper ];
+  buildInputs = [
+    # needed for GSETTINGS_SCHEMAS_PATH
+    gsettings-desktop-schemas glib gtk3
   ];
 
-  runtimeDependencies = [ (lib.getLib udev) libnotify gtk4 ];
-
-  desktopItems = [
-    (makeDesktopItem rec {
-      name = "Wavebox";
-      exec = "wavebox";
-      icon = "wavebox";
-      desktopName = name;
-      genericName = name;
-      categories = [ "Network" "WebBrowser" ];
-    })
-  ];
+  unpackPhase = ''
+    ar x $src
+    tar xf data.tar.xz
+  '';
+
+  rpath = lib.makeLibraryPath deps + ":" + lib.makeSearchPathOutput "lib" "lib64" deps;
+  binpath = lib.makeBinPath deps;
 
   installPhase = ''
     runHook preInstall
 
-    mkdir -p $out/bin $out/opt/wavebox
-    cp -r * $out/opt/wavebox
+    appname=wavebox
+
+    exe=$out/bin/$appname
+
+    mkdir -p $out/bin $out/share
+
+    cp -a opt/* $out/share
+    cp -a usr/share/* $out/share
 
-    # provide icon for desktop item
-    mkdir -p $out/share/icons/hicolor/128x128/apps
-    ln -s $out/opt/wavebox/product_logo_128.png $out/share/icons/hicolor/128x128/apps/wavebox.png
+
+    substituteInPlace $out/share/wavebox.io/$appname/$appname-launcher \
+      --replace 'CHROME_WRAPPER' 'WRAPPER'
+    substituteInPlace $out/share/applications/$appname.desktop \
+      --replace /opt/wavebox.io/wavebox/wavebox-launcher $exe
+    substituteInPlace $out/share/menu/$appname.menu \
+      --replace /opt $out/share \
+      --replace $out/share/wavebox.io/$appname/$appname $exe
+
+    for icon_file in $out/share/wavebox.io/wavebox/product_logo_[0-9]*.png; do
+      num_and_suffix="''${icon_file##*logo_}"
+      icon_size="''${num_and_suffix%.*}"
+      logo_output_prefix="$out/share/icons/hicolor"
+      logo_output_path="$logo_output_prefix/''${icon_size}x''${icon_size}/apps"
+      mkdir -p "$logo_output_path"
+      mv "$icon_file" "$logo_output_path/$appname.png"
+    done
+
+    makeWrapper "$out/share/wavebox.io/$appname/$appname" "$exe" \
+      --prefix LD_LIBRARY_PATH : "$rpath" \
```

Comment:
Setting LD_LIBRARY_PATH is highly discouraged especially with many libraries. Can this be reduced to the bare minimum instead of using a shotcun approach?

Reply:
Thanks for the detailed review and suggestions :bow: 

From v10.x onwards, Wavebox switched from an Electron app to a Chromium-derived browser. There were a number of issues regarding GPU acceleration in the packaging as a result of this change.

I reworked the Wavebox package based on a pretty much carbon copy of the Google Chrome package in Nixpkgs. Doing so resolved all the issues related to Wavebox finding GL-related libraries.

The use of `LD_LIBRARY_PATH` comes from the Google Chrome package. I have not investigated if it is required or if an alternative approach is suitable.